Dz_4_5/src/bohdan/homework/com/Fighters/Riders/DragonRider.java

```diff
@@ -1,51 +1,40 @@
 package bohdan.homework.com.Fighters.Riders;
 
 
+import bohdan.homework.com.Fighters.Base.Actions.FightBreakAction;
 import bohdan.homework.com.Fighters.Base.ArenaFighter;
 import bohdan.homework.com.Fighters.Dragons.Dragon;
-import bohdan.homework.com.Fighters.Riders.DragonPet.DragonPet;
 
 
-public class DragonRider extends ArenaFighter implements DragonPet{ //TODO: not over yet
+
+public class DragonRider extends ArenaFighter implements FightBreakAction {
 
     private Dragon dragonPet = null;
 
     public DragonRider(String name, int health, int damage, double defense) {
         super(name, health, damage, defense);
     }
 
-
-    public void dragonPET(Dragon dragon) {
-            dragonPet = dragon;
+    @Override
+    public void actionWithFight(ArenaFighter arenaFighter, FightCallBack fightCallBack) {
```

Comment:
if you use exceptions, first exceptions must be custom, Second  Fight callback should be removed. Instead, you have to "throws"  the custom exception to be handled outside this method.